src/metricscollector/server/memory_metric_handler_test.go

```diff
@@ -106,4 +115,194 @@ var _ = Describe("MemoryMetricHandler", func() {
 			})
 		})
 	})
+
+	Describe("GetMemoryMetricHistory", func() {
+		JustBeforeEach(func() {
+			handler.GetMemoryMetricHistory(resp, req, map[string]string{"appid": "an-app-id"})
+		})
+
+		Context("when request query string is invalid", func() {
+			Context("when there are multiple start pararmeters in query string", func() {
+				BeforeEach(func() {
+					req, err = http.NewRequest(http.MethodGet, "http://localhost/v1/apps/an-app-id/metrics_history/memory?start=123&start=231", nil)
+					Expect(err).ToNot(HaveOccurred())
+
+				})
+
+				It("returns 400", func() {
+					Expect(resp.Code).To(Equal(http.StatusBadRequest))
+
+					errJson := &ErrorResponse{}
+					d := json.NewDecoder(resp.Body)
+					err = d.Decode(errJson)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(errJson.Code).To(Equal("Bad-Request"))
```

Comment:
Why not just do a Equal(&ErrorResponse{
Code: "Bad-Request",
Message: ".....",
})